xyz-psql-connector/src/main/java/com/here/xyz/psql/SQLQueryBuilder.java

```diff
@@ -96,9 +99,7 @@ public static SQLQuery buildGetFeaturesByBBoxQuery(final GetFeaturesByBBoxEvent
         final SQLQuery geoQuery = new SQLQuery("ST_Intersects(geo, ST_MakeEnvelope(?, ?, ?, ?, 4326))",
                 bbox.minLon(), bbox.minLat(), bbox.maxLon(), bbox.maxLat());
 
-        boolean bConvertGeo2GeoJson = ( mvtTypeRequested(event) == 0 );
-
-        return generateCombinedQuery(event, geoQuery, searchQuery,dataSource, bConvertGeo2GeoJson, null);
+        return generateCombinedQuery(event, geoQuery, searchQuery,dataSource, getResponseType(event) == GEO_JSON, null);
```

Comment:
Each of the usages are happening locally inside different methods. With the current structure it will be hard to overcome that. Maybe that could be consolidated in further refactorings.